workflows/Cargo.toml

```diff
@@ -29,7 +30,13 @@ rand.workspace = true
 log.workspace = true
 eyre.workspace = true
 
+# system info
+sysinfo = "0.32.0"
```

Comment:
add this as an optional dependency, and feature-gate it behind a feature called `profiling`